vendor/bat-native-ads/src/bat/ads/internal/purchase_intent/keywords.cc

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <stdint.h>
+#include <algorithm>
+#include <sstream>
+
+#include "url/gurl.h"
+#include "third_party/re2/src/re2/re2.h"
+#include "bat/ads/internal/purchase_intent/keywords.h"
+
+namespace ads {
+
+Keywords::Keywords() = default;
+Keywords::~Keywords() = default;
+
+PurchaseIntentSegmentList Keywords::GetSegments(
+    const std::string& search_query) {
+  PurchaseIntentSegmentList segment_list;
+  auto search_query_keyword_set = TransformIntoSetOfWords(search_query);
+
+  for (const auto& keyword : _automotive_segment_keywords) {
+    auto list_keyword_set = TransformIntoSetOfWords(keyword.keywords);
+
+    if (Keywords::IsSubset(search_query_keyword_set, list_keyword_set)) {
+      segment_list = keyword.segments;
+      return segment_list;
+    }
+  }
+
+  return segment_list;
+}
+
+uint16_t Keywords::GetFunnelWeight(
+    const std::string& search_query) {
+  auto search_query_keyword_set = TransformIntoSetOfWords(search_query);
+
+  for (const auto& keyword : _automotive_funnel_keywords) {
+    auto list_keyword_set = TransformIntoSetOfWords(keyword.keywords);
+
+    if (Keywords::IsSubset(search_query_keyword_set, list_keyword_set)) {
+      return keyword.weight;
+    }
+  }
+
+  return _default_signal_weight;
+}
+
+// TODO(Moritz Haller): Rename to make explicit that method checks set_a is
+// subset of set_b but not vice versa
+bool Keywords::IsSubset(
+    std::vector<std::string> keyword_set_a,
+    std::vector<std::string> keyword_set_b) {
+  std::sort(keyword_set_a.begin(), keyword_set_a.end());
```

Comment:
Is it possible to use `std::set` instead, if  it doesn't break interface agreement? We don't need to sort it then.

Reply:
@gdregalo I added the existing ticket as a comment. Set impl is planned to come with the next PR as well. Are you happy with that or should we change for this PR?

`// TODO(https://github.com/brave/brave-browser/issues/8495): Implement Brave Ads Purchase Intent keyword matching with std::sets`